### PR TITLE
fix: drawer can not ref form instance when open

### LINKED ISSUE #35703: Drawer can not ref form instance when open
### Reproduction link

[https://stackblitz.com/edit/react-qnfrst?file=demo.js,package.json](https://stackblitz.com/edit/react-qnfrst?file=demo.js,package.json)

### Steps to reproduce

Open & see console

### What is expected?

As <= 4.20.5
https://stackblitz.com/edit/react-u6eetg?file=demo.js,package.json

### What is actually happening?

Empty ref

| Environment | Info |
| --- | --- |
| antd | 4.20.6 |
| React | latest |
| System | IOS |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #35703

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix drawer can not ref form instance when open     |
| 🇨🇳 Chinese |      修复抽屉打开时 form 实例  为 null  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by crazyair** [comment]
#35703

**Comment by zombieJ** [comment]
cc @MadCcc  check the CI bundle size

File: components/drawer/__tests__/DrawerEvent.test.js [hunk 1]
```diff
@@ -77,4 +78,13 @@ describe('Drawer', () => {
...
     expect(afterVisibleChange).toBeCalledTimes(1);
   });
+  it('should support form ref', () => {
+    const formRef = React.createRef();
+    render(
+      <Drawer visible>
+        <Form ref={formRef} />
+      </Drawer>,
+    );
+    expect(typeof formRef.current).toBe('object');
```
Comment (zombieJ) [inline components/drawer/__tests__/DrawerEvent.test.js:88]: 这个测试始终会过的。需要类似 #35703 的 Demo 对 Effect 下 ref 做判断。

Comment (crazyair) [reply components/drawer/__tests__/DrawerEvent.test.js]: 已修改

File: components/drawer/__tests__/DrawerEvent.test.js [hunk 2]
```diff
@@ -1,6 +1,27 @@
 import React from 'react';
 import Drawer from '..';
-import { render, fireEvent } from '../../../tests/utils';
+import { render, fireEvent, screen } from '../../../tests/utils';
+import Button from '../../button';
+
+const RefDemo = ({ fn }) => {
```
Comment (zombieJ) [inline components/drawer/__tests__/DrawerEvent.test.js:6]: 这个 Demo 直接写到下面的 it 里，只有它用，不要离测试太远了

Comment (crazyair) [reply components/drawer/__tests__/DrawerEvent.test.js]: 改了

File: components/drawer/__tests__/DrawerEvent.test.js [hunk 3]
```diff
@@ -77,4 +78,35 @@ describe('Drawer', () => {
...
+      return (
+        <>
+          <Button onClick={() => setVisible(true)}>open</Button>
+          <Drawer visible={visible}>
+            <div ref={ref} />
+          </Drawer>
+        </>
+      );
+    };
+    render(<RefDemo />);
```
Comment (zombieJ) [inline components/drawer/__tests__/DrawerEvent.test.js:108]: const { container } = render(<RefDemo />);
fireEvent.click(container.querySelector('button'));

**Pull request merged**